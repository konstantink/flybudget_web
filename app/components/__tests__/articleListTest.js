import React from 'react';
// import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ArticleList from '../ArticleList';

configure({adapter: new Adapter()});

describe('ArticleList', () =>{
  const testProps = {
    articles: {
      article_A: {
        id: 'article_A'
      },
      article_B: {
        id: 'article_B'
      }
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );

    // console.log(wrapper.getElement().props.children);
    expect(wrapper.getElement().props.children.length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
