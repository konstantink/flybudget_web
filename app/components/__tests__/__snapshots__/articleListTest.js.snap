// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "article_A": Object {
          "id": "article_A",
        },
        "article_B": Object {
          "id": "article_B",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArticleContainer
          article={
            Object {
              "id": "article_A",
            }
          }
        />,
        <ArticleContainer
          article={
            Object {
              "id": "article_B",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "article_A",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "id": "article_A",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "article_B",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "id": "article_B",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArticleContainer
            article={
              Object {
                "id": "article_A",
              }
            }
          />,
          <ArticleContainer
            article={
              Object {
                "id": "article_B",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "article_A",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "id": "article_A",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "article_B",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "id": "article_B",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
